### DISCUSSION TITLE
flutter build apk --split-per-abi --no-tree-shake-icons

### DISCUSSION THREAD

**Discussion by Hatalabdallah** [community/community#84186] [Mobile]
### Select Topic Area

General

### Body

PS C:\eBroker v1.1.0> flutter build apk --split-per-abi --no-tree-shake-icons
   
Building with Flutter multidex support enabled.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_sim_country_code-0.1.2\android\src\main\java\com\example\flutter_sim_country_code\FlutterSimCountryCodePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\firebase_messaging-14.7.9\android\src\main\java\io\flutter\plugins\firebase\messaging\FlutterFirebaseMessagingPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\firebase_dynamic_links-5.4.8\android\src\main\java\io\flutter\plugins\firebase\dynamiclinks\Utils.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
WARNING: [Processor] Library 'C:\Users\Jalaludeen\.gradle\caches\modules-2\files-2.1\me.carda\AndroidAwnCore\0.7.6\8c884ba6ca628f0483f85021a35111d9cf38e5ce\AndroidAwnCore-0.7.6.aar' contains references to both AndroidX and old support library. This seems like the library is partially migrated. Jetifier will try to rewrite the library anyway.
 Example of androidX reference: 'androidx/annotation/NonNull'
 Example of support library reference: 'android/support/v4/media/session/MediaSessionCompat'
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\flutter_native_image-0.0.6+1\android\src\main\java\com\example\flutternativeimage\FlutterNativeImagePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\geocoding_android-2.1.2\android\src\main\java\com\baseflow\geocoding\Geocoding.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\Users\Jalaludeen\AppData\Local\Pub\Cache\hosted\pub.dev\launch_review-3.0.1\android\src\main\java\com\iyaffle\launchreview\LaunchReviewPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Parameter format not correct -
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
w: Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk8/1.8.22/b25c86d47d6b962b9cf0f8c3f320c8a10eea3dd1/kotlin-stdlib-jdk8-1.8.22.jar (version 1.8)
    C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-jdk7/1.8.22/4dabb8248310d833bb6a8b516024a91fd3d275c/kotlin-stdlib-jdk7-1.8.22.jar (version 1.8)
    C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.0/8ee15ef0c67dc83d874f412d84378d7f0eb50b63/kotlin-stdlib-1.9.0.jar (version 1.9)
    C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.9.0/cd65c21cfd1eec4d44ef09f9f52b6d9f8a720636/kotlin-stdlib-common-1.9.0.jar (version 1.9)
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
e: C:/eBroker v1.1.0/build/wakelock/.transforms/0626f91f281117c931a06a027d166860/transformed/out/jars/classes.jar!/META-INF/wakelock_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/motion_sensors/.transforms/5685935999f962bc68b8beabb4406f45/transformed/out/jars/classes.jar!/META-INF/motion_sensors_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/share_plus/.transforms/3e052cf10325c81d93094a47099afd31/transformed/out/jars/classes.jar!/META-INF/share_plus_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/fluttertoast/.transforms/0ff8be71ce82a05d3c8cdbb81657843f/transformed/out/jars/classes.jar!/META-INF/fluttertoast_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/flutter_paystack/.transforms/616a36d63032cf5eec3a9ee335048668/transformed/out/jars/classes.jar!/META-INF/flutter_paystack_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/audioplayers_android/.transforms/ce08a10dfbf626fce0f38984a77adc51/transformed/out/jars/classes.jar!/META-INF/audioplayers_android_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/package_info_plus/.transforms/9c6c7b9a07ac6153a1cde7a0fd49eb96/transformed/out/jars/classes.jar!/META-INF/package_info_plus_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/eBroker v1.1.0/build/stripe_android/.transforms/0fb125eeea1fcc0b71e0b501f3385363/transformed/out/jars/classes.jar!/META-INF/stripe_android_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/27592ef912660bcc023e876d3cc2ce0e/transformed/jetified-play-services-measurement-api-21.5.0/jars/classes.jar!/META-INF/java.com.google.android.gmscore.integ.client.measurement_api_measurement_api.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/1054aeb56b6bb48276f25cb768675714/transformed/jetified-firebase-installations-17.2.0/jars/classes.jar!/META-INF/com.google.firebase-firebase-installations.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.     
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/b6235a2b5862447e8c6103e794ff3b31/transformed/jetified-firebase-common-ktx-20.4.2/jars/classes.jar!/META-INF/com.google.firebase-firebase-common-ktx.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/876870c5f85335274d9e6e2b03dd1053/transformed/jetified-firebase-common-20.4.2/jars/classes.jar!/META-INF/com.google.firebase-firebase-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/b991c0eb33607eb4195897c96e67135f/transformed/jetified-ads-adservices-java-1.0.0-beta05/jars/classes.jar!/META-INF/ads-adservices-java_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/a194c1545e46237a107ce6de104f8db7/transformed/jetified-ads-adservices-1.0.0-beta05/jars/classes.jar!/META-INF/ads-adservices_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlinx/kotlinx-coroutines-core-jvm/1.7.1/63a0779cf668e2a47d13fda7c3b0c4f8dc7762f4/kotlinx-coroutines-core-jvm-1.7.1.jar!/META-INF/kotlinx-coroutines-core.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlinx/kotlinx-coroutines-play-services/1.7.1/6333bad6f256e2ca7bc2908f586be7161a41618c/kotlinx-coroutines-play-services-1.7.1.jar!/META-INF/kotlinx-coroutines-play-services.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/f24894e073d0cc6344eb1080e9b51564/transformed/jetified-activity-1.7.2/jars/classes.jar!/META-INF/activity_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/f169ec4362f00aa9717e000a7387c1f9/transformed/lifecycle-livedata-2.6.1/jars/classes.jar!/META-INF/lifecycle-livedata_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/4b86ca06f88f06a3281547c5e5a170d5/transformed/lifecycle-viewmodel-2.6.1/jars/classes.jar!/META-INF/lifecycle-viewmodel_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/a5e9470d997a4c743f92ca5e38c7333e/transformed/lifecycle-livedata-core-2.6.1/jars/classes.jar!/META-INF/lifecycle-livedata-core_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/7b0aced55018b41b3a2cdd84d22958dc/transformed/jetified-lifecycle-viewmodel-savedstate-2.6.1/jars/classes.jar!/META-INF/lifecycle-viewmodel-savedstate_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.  
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/de02b991e12c5276de424497371dccb6/transformed/jetified-core-ktx-1.10.1/jars/classes.jar!/META-INF/core-ktx_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/fc09af33fd8a999cb6157eddae2bd577/transformed/core-1.10.1/jars/classes.jar!/META-INF/core_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/400c6dfc953f68ca937f450ebb34d2e8/transformed/lifecycle-runtime-2.6.1/jars/classes.jar!/META-INF/lifecycle-runtime_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/androidx.lifecycle/lifecycle-common/2.6.1/10f354fdb64868baecd67128560c5a0d6312c495/lifecycle-common-2.6.1.jar!/META-INF/lifecycle-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlinx/kotlinx-coroutines-android/1.7.1/c2d86b569f10b7fc7e28d3f50c0eed97897d77a7/kotlinx-coroutines-android-1.7.1.jar!/META-INF/kotlinx-coroutines-android.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/93180f7047a1b15cf5285ad5abed971e/transformed/jetified-savedstate-1.2.1/jars/classes.jar!/META-INF/savedstate_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/androidx.annotation/annotation-jvm/1.7.1/920472d40adcdef5e18708976b3e314f9a636fcd/annotation-jvm-1.7.1.jar!/META-INF/annotation.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/transforms-3/440c2f4ab6f100e4ef64212016268083/transformed/jetified-annotation-experimental-1.3.0/jars/classes.jar!/META-INF/annotation-experimental_release.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.0/8ee15ef0c67dc83d874f412d84378d7f0eb50b63/kotlin-stdlib-1.9.0.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.0/8ee15ef0c67dc83d874f412d84378d7f0eb50b63/kotlin-stdlib-1.9.0.jar!/META-INF/kotlin-stdlib-jdk8.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.0/8ee15ef0c67dc83d874f412d84378d7f0eb50b63/kotlin-stdlib-1.9.0.jar!/META-INF/kotlin-stdlib-jdk7.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.5.1.
e: C:/Users/Jalaludeen/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.9.0/cd65c21cfd1eec4d44ef09f9f52b6d9f8a720636/kotlin-stdlib-common-1.9.0.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.5.1. 
ERROR:R8: java.lang.OutOfMemoryError: Java heap space

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> com.android.tools.r8.CompilationFailedException: Compilation failed to complete        

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 48m 16s

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':app:minifyReleaseWithR8'.
> com.android.tools.r8.CompilationFailedException: Compilation failed to complete        

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
Java heap space

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 48m 18s

FAILURE: Build failed with an exception.

* What went wrong:
Could not receive a message from the daemon.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org
Running Gradle task 'assembleRelease'...                         2905.4s

┌─ Flutter Fix ─────────────────────────────────────────────────────────────────────────┐│ [!] Your project requires a newer version of the Kotlin Gradle plugin.                ││ Find the latest version on https://kotlinlang.org/docs/releases.html#release-details, ││ then update C:\eBroker v1.1.0\android\build.gradle:                                   ││ ext.kotlin_version = '<latest-version>'                                               │└───────────────────────────────────────────────────────────────────────────────────────┘Gradle task assembleRelease failed with exit code 1
PS C:\eBroker v1.1.0>

**Comment by esahmed21** [comment]
Hi @Hatalabdallah 👋 

Thank you for reaching out, We appreciate your engagement with the community.

However, the focus of this community discussion is centered around our GitHub app. If you have any questions or concerns related to the app, please feel free to share them, and we'll be happy to assist.

For topics unrelated to our app, we recommend exploring other forums or communities that specialize in those areas.

Thanks for your understanding and cooperation!

**Comment by devbeda** [comment]
hey i am also facing same error can you guide me how can you resolved it

**Comment by HenderRicard** [comment]
Hender2809@&

**Comment by miabilal** [comment]
@Hatalabdallah  does your issue solved if not then try to update your kotlin version in the android/build.gradle
